src/main/java/org/mule/extension/aggregator/internal/storage/content/SimpleAggregatedContent.java

```diff
@@ -25,7 +25,7 @@
 public class SimpleAggregatedContent extends AbstractAggregatedContent {
 
   private static final long serialVersionUID = -229638907750317297L;
-  private Map<Integer, TypedValue> sequencedElements;
+  private Map<Integer, List<TypedValue>> sequencedElements;
```

Comment:
It has to be forward compatible. The new connector has to know how to read the information saved by the previous version. As @estebanwasinger pointed out.